packages/discovery-provider/src/queries/get_user_purchased_content.py

```diff
@@ -0,0 +1,118 @@
+
+    user_id = args.get("user_id")
+
+    query = (
+        session.query(USDCPurchase, Track, Playlist)
+        .outerjoin(
+            Track,
+            and_(
+                USDCPurchase.content_type == "track",
+                USDCPurchase.content_id == Track.track_id,
+                Track.is_current == True,
+                Track.is_delete == False,
+                Track.is_unlisted == False,
+            ),
+        )
+        .outerjoin(
+            Playlist,
+            and_(
+                USDCPurchase.content_type == "album",
+                USDCPurchase.content_id == Playlist.playlist_id,
+                Playlist.is_current == True,
+                Playlist.is_delete == False,
+                Playlist.is_private == False,
+            ),
+        )
+        .distinct(Track.track_id, Playlist.playlist_id)
+        .filter(
+            USDCPurchase.seller_user_id == user_id,
+            # Drop rows that have no join found for either track or playlist
+            or_(Track.track_id != None, Playlist.playlist_id != None),
+            or_(Track.owner_id == user_id, Playlist.playlist_owner_id == user_id),
+        )
+        .order_by(
+            Track.track_id.desc(),
+            Playlist.playlist_id.desc(),
+            USDCPurchase.created_at.desc(),
+            USDCPurchase.content_id.desc(),
+            USDCPurchase.content_type.desc(),
+        )
+    )
+
+    return query
+
+
+def get_user_purchasable_content(args: GetUserPurchasedContentArgs):
+    user_purchasable_content = []
+    current_user_id = args.get("current_user_id")
+    limit = args.get("limit")
+    offset = args.get("offset")
+    db = get_db_read_replica()
+
+    with db.scoped_session() as session:
+        query = _get_user_purchasable_content(session, args)
+        query_results = add_query_pagination(query, limit, offset).all()
+
+        tracks = helpers.query_result_to_list(
+            filter(None, [item[1] for item in query_results])
+        )
+        playlists = helpers.query_result_to_list(
+            filter(None, [item[2] for item in query_results])
+        )
+
+        track_ids = [track["track_id"] for track in tracks]
+        tracks = populate_track_metadata(session, track_ids, tracks, current_user_id)
+        playlist_ids = [playlist["playlist_id"] for playlist in playlists]
+        playlists = populate_playlist_metadata(
+            session,
+            playlist_ids,
+            playlists,
+            [RepostType.playlist, RepostType.album],
+            [SaveType.playlist, SaveType.album],
+            current_user_id,
+        )
+
+        unsorted_purchasable_content = tracks + playlists
+        unsorted_purchasable_content = [
+            # The user gets wrapped in an array due to the joins, so we need to unpack it
+            {**content, "user": content["user"][0]}
+            for content in unsorted_purchasable_content
+        ]
+
+        user_purchasable_content = sorted(
```

Comment:
Does/should the api accept a sorting parameter? Or is it just always reverse chronological?